Automattic/studio
# Studio:  Implement disable chat UI when the user reaches prompt limit

Closes https://github.com/Automattic/dotcom-forge/issues/7761

## Proposed Changes

This PR adds UI for the case when the user has reached the limit of prompts. When the chat is new:

<img width="914" alt="Screenshot 2024-06-18 at 3 49 45 PM" src="https://github.com/Automattic/studio/assets/25575134/4c2385bb-6a4c-40be-8235-de4d1af9f618">

When the user has previous messages in chat:

<img width="898" alt="Screenshot 2024-06-18 at 3 49 38 PM" src="https://github.com/Automattic/studio/assets/25575134/805e5cff-39c0-45da-9132-59f1246f0c0e">

## Testing Instructions

* Pull the changes from this branch locally
* Start the app with the AI Assistant: `STUDIO_AI=true npm start`
* Apply the following diff:

```diff --git a/src/hooks/use-prompt-usage.tsx b/src/hooks/use-prompt-usage.tsx
index 282b02a..9a119e1 100644
--- a/src/hooks/use-prompt-usage.tsx
+++ b/src/hooks/use-prompt-usage.tsx
@@ -46,7 +46,7 @@ export function PromptUsageProvider( { children }: PromptUsageProps ) {
                                return;
                        }
                        setPromptLimit( limit );
-                       setPromptCount( limit - remaining );
+                       setPromptCount( limit );
                        if ( ! initiated ) {
                                setInitiated( true );
                        }
```

* Navigate to the AI Assistant
* Confirm that you see the message that the chat is disabled along with the count of days in which your input will be reset
* If it is a new chat, confirm that no Welcome Component is disabled
* If it is an old chat, confirm that you can see previous messages but can't send a new message

## Pre-merge Checklist
